### PR TITLE
chore(docs): general docs and types cleanup

### PR DESCRIPTION
**Pull request by amcdnl**
This PR aims to:

- Create consistency with typings
- Add documentation to missing properties/methods
- Exclude private methods that can't be actually private from docs
- Fix bad descriptions on methods

### CODE REVIEW

File: src/cdk/table/table.ts [hunk 1]
```diff
@@ -169,18 +169,18 @@ export class CdkTable<T> implements CollectionViewer {
...
 
-  ngAfterContentInit() {
+  ngAfterContentInit(): void {
     this._cacheColumnDefsByName();
     this._columnDefs.changes.subscribe(() => this._cacheColumnDefsByName());
     this._renderHeaderRow();
   }
 
-  ngAfterContentChecked() {
+  ngAfterContentChecked(): void {
```
Comment (DevVersion) [inline src/cdk/table/table.ts]: Lifecycle hooks are not necessary. They are being excluded by our dgeni processors.

File: src/cdk/table/table.ts [hunk 2]
```diff
@@ -218,7 +218,7 @@ export class CdkTable<T> implements CollectionViewer {
    * Check if the header or rows have changed what columns they want to display. If there is a diff,
    * then re-render that section.
    */
-  private _renderUpdatedColumns() {
+  private _renderUpdatedColumns(): void {
```
Comment (DevVersion) [inline src/cdk/table/table.ts]: Private methods in general don't need to have an explicit type.

File: src/cdk/table/table.ts (hunk 1 above)
Comment (amcdnl) [reply src/cdk/table/table.ts]: I'm trying to create consistency on typing. If everything else has it, we might as well make these have it too.

File: src/cdk/table/table.ts (hunk 2 above)
Comment (amcdnl) [reply src/cdk/table/table.ts]: I'm trying to create consistency on typing.

Comment (crisbeto) [reply src/cdk/table/table.ts]: The convention at the moment is that if it goes into the docs, it should be typed (so if it's public, doesn't start with an underscore and it's not an Angular lifecycle hook). IMO having to add the types everywhere might cause some churn.

**Comment by amcdnl** [comment]
Given the feedback, I'm going to close this and only go back and type things that will show up in the docs.

**Pull request closed without merging**